.github/workflows/CI.yml

```diff
@@ -62,7 +62,7 @@ jobs:
     # Run the job for different versions of python
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
This is fine for the moment, because we're still using 3.9 in deployment, but when we drop that we should also remove this from the CI so we don't just keep adding versions. Three seems like a decent span to test against.

Reply:
Ok, will do that once this issue updating to 3.10 is closed (https://github.com/ACCESS-NRI/payu-condaenv/issues/32)